frontend/routes/modelSettings/modelSettings.component.tsx

```diff
@@ -0,0 +1,390 @@
+/**
+ *  Copyright (C) 2017 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import * as React from 'react';
+import * as queryString from 'query-string';
+import { Formik, Field } from 'formik';
+import { snakeCase } from 'lodash';
+
+import Button from '@material-ui/core/Button';
+import Grid from '@material-ui/core/Grid';
+import InputLabel from '@material-ui/core/InputLabel';
+import Input from '@material-ui/core/Input';
+
+import { clientConfigService } from '../../services/clientConfig';
+
+import { Panel } from '../components/panel/panel.component';
+import { CellSelect } from '../components/customTable/components/cellSelect/cellSelect.component';
+
+import {
+	FieldsRow,
+	StyledTextField,
+	SelectWrapper,
+	StyledChip,
+	TopicTypesContainer,
+	StyledForm,
+	Headline,
+	TypesGrid,
+	GridColumn
+} from './modelSettings.styles';
+
+const PANEL_PROPS = {
+	title: 'Model Settings',
+	paperProps: {
+		height: '100%'
+	}
+};
+
+const ENTER_KEY = 'Enter';
```

Comment:
According to our conversation, you could move `ENTER_KEY` and `unitsMap` to new constants' files.